src/network/p2p/client.rs

```diff
@@ -296,22 +296,54 @@ impl Command for CountKademliaPeers {
 }
 
 struct GetLocalInfo {
+	peer_id: Option<PeerId>,
 	response_sender: Option<oneshot::Sender<Result<LocalInfo>>>,
 }
 
 impl Command for GetLocalInfo {
 	fn run(&mut self, entries: EventLoopEntries) -> Result<(), Report> {
+		let (peer_id, peer_multiaddr) = match self.peer_id {
```

Comment:
We should probably add a new command for this, in order to split the domains of peer vs external peer info.